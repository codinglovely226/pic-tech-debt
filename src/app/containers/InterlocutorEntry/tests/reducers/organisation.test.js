/* eslint-disable */

import locationReducer from '../../../CaseEntry/reducers/location';

import {
  updateInterlocutorPlaceId,
  updateInterlocutorAddress1,
  updateInterlocutorAddress2,
  updateInterlocutorCity,
  updateInterlocutorCountry,
  updateInterlocutorISOCountry,
  updateInterlocutorGeometry,
  updateInterlocutorLocation,
  updateInterlocutorOrganisation,
  updateInterlocutorPostcode,
  updateInterlocutorState,
  updateInterlocutorTelephone,
} from '../../actions';

describe('locationReducer', () => {
  it('returns the initial state', () => {
    expect(locationReducer(undefined, {})).toMatchSnapshot();
  });
  it('update caller place ID', () => {
    expect(locationReducer({}, updateInterlocutorPlaceId(1, 'Place ID'))).toMatchSnapshot();
  });
  it('update caller address 1', () => {
    expect(locationReducer({}, updateInterlocutorAddress1(1, 'Address line 1'))).toMatchSnapshot();
  });
  it('update caller address 2', () => {
    expect(locationReducer({}, updateInterlocutorAddress2(1, 'Address line 2'))).toMatchSnapshot();
  });
  it('update caller city', () => {
    expect(locationReducer({}, updateInterlocutorCity(1, 'Cairo'))).toMatchSnapshot();
  });
  it('update caller country', () => {
    expect(locationReducer({}, updateInterlocutorCountry(1, 'Slovenia'))).toMatchSnapshot();
  });
  it('update caller ISO country', () => {
    expect(locationReducer({}, updateInterlocutorISOCountry(1, 'AU'))).toMatchSnapshot();
  });
  it('update caller geometry', () => {
    expect(locationReducer({}, updateInterlocutorGeometry(1, {'lat': 1, 'lng': -1}))).toMatchSnapshot();
  });
  it('update caller location', () => {
    expect(locationReducer({}, updateInterlocutorLocation(1, 'Old location', 'Test location'))).toMatchSnapshot();
  });
  it('update caller organisation name', () => {
    expect(locationReducer({}, updateInterlocutorOrganisation(1, 'Test organisation'))).toMatchSnapshot();
  });
  it('update caller postcode', () => {
    expect(locationReducer({}, updateInterlocutorPostcode(1, '5000'))).toMatchSnapshot();
  });
  it('update caller state', () => {
    expect(locationReducer({}, updateInterlocutorState(1, 'Devonshire'))).toMatchSnapshot();
  });
  it('update caller telephone', () => {
    expect(locationReducer({}, updateInterlocutorTelephone(1, '(02) 9000 0000'))).toMatchSnapshot();
  });
});
