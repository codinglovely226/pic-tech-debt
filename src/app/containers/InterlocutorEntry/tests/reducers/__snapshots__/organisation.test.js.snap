exports[`locationReducer returns the initial state 1`] = `
Object {
  "address1": "",
  "address2": "",
  "city": "",
  "country": "",
  "geometry": Object {},
  "isoCountry": "",
  "location": "",
  "name": "",
  "placeId": "",
  "postcode": "",
  "state": "",
  "telephone": "",
}
`;

exports[`locationReducer update caller ISO country 1`] = `
Object {
  "address1": "",
  "address2": "",
  "city": "",
  "country": "",
  "geometry": Object {},
  "isoCountry": "AU",
  "location": "",
  "name": "",
  "placeId": "",
  "postcode": "",
  "state": "",
  "telephone": "",
}
`;

exports[`locationReducer update caller address 1 1`] = `
Object {
  "address1": "Address line 1",
  "address2": "",
  "city": "",
  "country": "",
  "geometry": Object {},
  "isoCountry": "",
  "location": "",
  "name": "",
  "placeId": "",
  "postcode": "",
  "state": "",
  "telephone": "",
}
`;

exports[`locationReducer update caller address 2 1`] = `
Object {
  "address1": "",
  "address2": "Address line 2",
  "city": "",
  "country": "",
  "geometry": Object {},
  "isoCountry": "",
  "location": "",
  "name": "",
  "placeId": "",
  "postcode": "",
  "state": "",
  "telephone": "",
}
`;

exports[`locationReducer update caller city 1`] = `
Object {
  "address1": "",
  "address2": "",
  "city": "Cairo",
  "country": "",
  "geometry": Object {},
  "isoCountry": "",
  "location": "",
  "name": "",
  "placeId": "",
  "postcode": "",
  "state": "",
  "telephone": "",
}
`;

exports[`locationReducer update caller country 1`] = `
Object {
  "address1": "",
  "address2": "",
  "city": "",
  "country": "Slovenia",
  "geometry": Object {},
  "isoCountry": "",
  "location": "",
  "name": "",
  "placeId": "",
  "postcode": "",
  "state": "",
  "telephone": "",
}
`;

exports[`locationReducer update caller geometry 1`] = `
Object {
  "address1": "",
  "address2": "",
  "city": "",
  "country": "",
  "geometry": Object {
    "lat": 1,
    "lng": -1,
  },
  "isoCountry": "",
  "location": "",
  "name": "",
  "placeId": "",
  "postcode": "",
  "state": "",
  "telephone": "",
}
`;

exports[`locationReducer update caller location 1`] = `
Object {
  "address1": "",
  "address2": "",
  "city": "",
  "country": "",
  "geometry": Object {},
  "isoCountry": "",
  "location": "Test location",
  "name": "",
  "placeId": "",
  "postcode": "",
  "state": "",
  "telephone": "",
}
`;

exports[`locationReducer update caller organisation name 1`] = `
Object {
  "address1": "",
  "address2": "",
  "city": "",
  "country": "",
  "geometry": Object {},
  "isoCountry": "",
  "location": "",
  "name": "Test organisation",
  "placeId": "",
  "postcode": "",
  "state": "",
  "telephone": "",
}
`;

exports[`locationReducer update caller place ID 1`] = `
Object {
  "address1": "",
  "address2": "",
  "city": "",
  "country": "",
  "geometry": Object {},
  "isoCountry": "",
  "location": "",
  "name": "",
  "placeId": "Place ID",
  "postcode": "",
  "state": "",
  "telephone": "",
}
`;

exports[`locationReducer update caller postcode 1`] = `
Object {
  "address1": "",
  "address2": "",
  "city": "",
  "country": "",
  "geometry": Object {},
  "isoCountry": "",
  "location": "",
  "name": "",
  "placeId": "",
  "postcode": "5000",
  "state": "",
  "telephone": "",
}
`;

exports[`locationReducer update caller state 1`] = `
Object {
  "address1": "",
  "address2": "",
  "city": "",
  "country": "",
  "geometry": Object {},
  "isoCountry": "",
  "location": "",
  "name": "",
  "placeId": "",
  "postcode": "",
  "state": "Devonshire",
  "telephone": "",
}
`;

exports[`locationReducer update caller telephone 1`] = `
Object {
  "address1": "",
  "address2": "",
  "city": "",
  "country": "",
  "geometry": Object {},
  "isoCountry": "",
  "location": "",
  "name": "",
  "placeId": "",
  "postcode": "",
  "state": "",
  "telephone": "(02) 9000 0000",
}
`;
